OpenSim/Common/DelimFileAdapter.h

```diff
@@ -317,6 +317,13 @@ DelimFileAdapter<T>::extendRead(const std::string& fileName) const {
     std::string line{};
     while(std::getline(in_stream, line)) {
         ++line_num;
+
+        // We might be parsing a file with CRLF (\r\n) line endings on a
+        // platform that uses only LF (\n) line endings, in which case the \r
+        // is part of `line` and we must remove it manually.
+        if (line.size() && line[line.size() - 1] == '\r') 
+            line = line.substr(0, line.size() - 1);
```

Comment:
Not sure, but I think a new string will be constructed here. Would `line.pop_back();` be more efficient? Speed might actually matter here if the file is long.